scripts/hostapd/install

```diff
@@ -27,10 +27,10 @@ echo "iface lo inet loopback" | sudo tee --append /etc/network/interfaces > /dev
 echo "" | sudo tee --append /etc/network/interfaces > /dev/null
 echo "allow-hotplug wlan0" | sudo tee --append /etc/network/interfaces > /dev/null
 echo "iface wlan0 inet static" | sudo tee --append /etc/network/interfaces > /dev/null
-echo "    address 10.0.0.1" | sudo tee --append /etc/network/interfaces > /dev/null
-echo "    netmask 255.255.255.0" | sudo tee --append /etc/network/interfaces > /dev/null
-echo "    network 10.0.0.0" | sudo tee --append /etc/network/interfaces > /dev/null
-echo "    broadcast 10.0.0.255" | sudo tee --append /etc/network/interfaces > /dev/null
+echo "    address 10.10.0.1" | sudo tee --append /etc/network/interfaces > /dev/null
+echo "    netmask 255.255.255.128" | sudo tee --append /etc/network/interfaces > /dev/null
```

Comment:
Setting to such a non-standard net mask is not really needed. it does not have to match the dhcp scope we could just leave it as 255.255.255.0

Reply:
But we discussed this on chat, to use `255.255.255.128` to match DHCP scope, no? Have I misunderstood?

Reply:
TL;DR - Range NEVER matches the netmask, its always a smaller subset of iwhat the netmask defines.

So again the config IS correct but not needed (kinda see below for a correction)

DHCP SCOPE does not need to "match" the available network addresses, in fact its usually a small subset of the available range, and never ALL of it (because the dhcp server needs an ip (ie 10.0.0.1))

The only limit is that the scope MUST BE contained within the range of the network

```dhcp-range=10.0.0.2,10.0.0.127,255.255.255.0,12h ```
This example asks dhcp to issue ip addresses between 2 to 127 with the netmask of 255.255.255.0. In such a network you can have ips all the way up to 254 but DHCP simply wont offer them up. This is usually reserved for things like STATIC IP ADDRESSES for a printer, or server or other.


SIDE NOTE
> Something i have seen with "public" wifi networks is that they dont make the scope big enough and the expiry time short enough. So in this example the first 127 people that join the network will work fine the 128 person even though he/she joined 9 hours after everyone else left, wont get on because there are no more ip addresses left to offer until the old ones expire. This is not a problem with out setup but in a roaming environment it may be.

Now Reading my  previous comment, I think there is an error in this line actually
```dhcp-range=10.0.0.2,10.0.0.126,12h ```

This would let the computer define its own subnet, and as we saw in the previous comment the subset would be assigned by default as 10.0.0.0/16

You did not notice this problem because although the subnets the overlap they don't do it on the router but rather  on your laptop (or other device). Because your laptop does not know of  10.10.0.0/24 (its not one of the interfaces there) it "doesnt matter"

The issue would arise if you tried to (for example) ping 10.10.0.1
your computer would think that 10.10.0.1 was on the SAME LAN as 10.0.0.0 because of the incorrect NETMASK of 255.0.0.0 and try to switch it instead of routing it and it would never get to where its going.

Reply:
The laptop does pick up the subnet mask from **/etc/network/interfaces** even when I don't specify it on the mask in `dhcp-range` in **dnsmasq.conf**:

![screen shot 2017-12-22 at 2 11 40 pm](https://user-images.githubusercontent.com/2261308/34299222-1438988c-e722-11e7-902f-d03c45282a17.png)

But here I have `dhcp-range=10.10.0.2,10.10.0.126,12h` not `10.0`.

Reply:
ok cool... looks like dnsmasq already pulls it from the interface and not the ancient "class a , b ,c " definition :+1:

Reply:
I will add in dnsmasq anyways so there is no ambiguity